serialization/src/hex_encoded.rs

```diff
@@ -0,0 +1,67 @@
+// Copyright (c) 2023 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://github.com/mintlayer/mintlayer-core/blob/master/LICENSE
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::str::FromStr;
+
+use crate::hex::{HexDecode, HexEncode, HexError};
+
+/// Wrapper that serializes objects as hex encoded string for `serde`
+#[derive(Debug, Clone)]
+pub struct HexEncoded<T>(T);
+
+impl<T> HexEncoded<T> {
+    pub fn new(value: T) -> Self {
+        Self(value)
+    }
+
+    pub fn into_inner(self) -> T {
```

Comment:
I would call this `take`

Reply:
OK, renamed.
But I think the `take` name is usually used with mutable references:
https://doc.rust-lang.org/std/mem/fn.take.html
https://doc.rust-lang.org/std/option/enum.Option.html#method.take
And `into_inner` when `self` is consumed:
https://doc.rust-lang.org/stable/std/boxed/struct.Box.html#method.into_inner

Reply:
I see. Actually we do use `take` in WithId too. If we want to change this we have to fix it there too.